flink-libraries/flink-cep/src/main/java/org/apache/flink/cep/nfa/DeweyNumber.java

```diff
@@ -175,16 +175,18 @@ public String toString() {
 	public static DeweyNumber fromString(final String deweyNumberString) {
 		String[] splits = deweyNumberString.split("\\.");
 
-		if (splits.length == 0) {
+		if (splits.length == 1) {
 			return new DeweyNumber(Integer.parseInt(deweyNumberString));
-		} else {
+		} else if (splits.length > 0) {
```

Comment:
In which case would the returned `splits` has a length not larger than `0`? I think the condition `if (splits.length > 0)` might be redundant.

Reply:
@Myasuka thanks for the comments. 
In some rare cases, the length can be 0. So this condition guards against such cases.